pkg/gitprovider/gitee.go

```diff
@@ -0,0 +1,380 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package gitprovider
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"encoding/json"
+
+	"gitee.com/openeuler/go-gitee/gitee"
+	"github.com/antihax/optional"
+	"golang.org/x/oauth2"
+)
+
+type Project struct {
+	Id            int32            `json:"id,omitempty"`
+	FullName      string           `json:"full_name,omitempty"`
+	HumanName     string           `json:"human_name,omitempty"`
+	Url           string           `json:"url,omitempty"`
+	Namespace     *gitee.Namespace `json:"namespace,omitempty"`
+	Path          string           `json:"path,omitempty"`
+	Name          string           `json:"name,omitempty"`
+	Owner         *gitee.UserBasic `json:"owner,omitempty"`
+	Description   string           `json:"description,omitempty"`
+	Private       bool             `json:"private,omitempty"`
+	Public        bool             `json:"public,omitempty"`
+	Internal      bool             `json:"internal,omitempty"`
+	HtmlUrl       string           `json:"html_url,omitempty"`
+	Homepage      string           `json:"homepage,omitempty"`
+	DefaultBranch string           `json:"default_branch,omitempty"`
+	Parent        *Project         `json:"parent,omitempty"`
+}
+
+type GiteeGitProvider struct {
+	*AbstractGitProvider
+
+	token string
+}
+
+func NewGiteeGitProvider(token string) *GiteeGitProvider {
+	gitProvider := &GiteeGitProvider{
+		token:               token,
+		AbstractGitProvider: &AbstractGitProvider{},
+	}
+	gitProvider.AbstractGitProvider.GitProvider = gitProvider
+
+	return gitProvider
+}
+
+func (g *GiteeGitProvider) CanHandle(repoUrl string) (bool, error) {
+	staticContext, err := g.ParseStaticGitContext(repoUrl)
+	if err != nil {
+		return false, err
+	}
+
+	return strings.Contains("https://gitee.com", staticContext.Source), nil
+}
+
+func (g *GiteeGitProvider) getApiClient() *gitee.APIClient {
+	tokenSource := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: g.token})
+	conf := gitee.NewConfiguration()
+	conf.HTTPClient = oauth2.NewClient(context.Background(), tokenSource)
+	return gitee.NewAPIClient(conf)
+}
+
+func (g *GiteeGitProvider) GetUser() (*GitUser, error) {
+	client := g.getApiClient()
+	ctx := context.Background()
+	user, _, err := client.UsersApi.GetV5User(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch User : %w", err)
+	}
+
+	gitUser := &GitUser{
+		Id:       strconv.FormatInt(int64(user.Id), 10),
+		Username: user.Login,
+		Name:     user.Name,
+	}
+
+	// gitee email will be empty if not set to public
+	if user.Email != "" {
+		gitUser.Email = user.Email
+	}
```

Comment:
@Tpuljak, we get an empty email if the email is not set to public in the profile. Should I add a note in the docs about making the email public?